frontend/src/hooks/useLogin.js

```diff
@@ -1,39 +1,37 @@
 import { useState } from "react";
-import { useAuthContext } from "./useAuthContext";
+import { useAuthContext } from './useAuthContext'
 
 export const useLogin = () => {
-  const [error, setError] = useState(null);
-  const [isLoading, setIsLoading] = useState(null);
-  const { dispatch } = useAuthContext();
-
-  const login = async (email, password) => {
-    setIsLoading(true);
-    setError(null);
-
-    const response = await fetch("/api/user/login", {
-      method: "POST",
-      headers: { "Content-Type": "application/json" },
-      body: JSON.stringify({ email, password }),
-    });
-
-    const json = await response.json();
-
-    if (!response.ok) {
-      setIsLoading(false);
-      setError(json.error);
-    }
-
-    if (response.ok) {
-      // Saving the user to localStorage
-      localStorage.setItem("user", JSON.stringify(json));
-
-      // Updating the auth context
-      dispatch({ type: "LOGIN", payload: json });
-
-      // Update loading state
-      setIsLoading(false);
+    const [error, setError] = useState(null)
+    const [isLoading, setIsLoading] = useState(null)
+    const { dispatch } = useAuthContext()
+
+    const login = async (email, password) => {
+        setIsLoading(true)
+        setError(null)
+
+        const response = await fetch('http://localhost:5000/api/user/login', {
```

Comment:
Use .env file and add the URL in env file. Easy to change later when you are deploying the application.